backend/llm/clients/openrouter.py

```
@@ -0,0 +1,111 @@
+                    raise LLMRateLimitError("OpenRouter rate limit exceeded") from error
+
+                error_body = error.read().decode("utf-8", errors="replace")
+                raise LLMProviderError(
+                    f"OpenRouter request failed: {error_body}",
+                    status_code=error.code,
+                ) from error
+            except (urllib.error.URLError, TimeoutError, socket.timeout) as error:
+                if attempt < self.max_retries:
+                    time.sleep(2**attempt)
+                    continue
+                raise LLMTimeoutError("OpenRouter request timed out") from error
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)